pkg/operator/k8sutil/labels.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sutil
+
+import (
+	"strings"
+)
+
+// ParseStringToLabels parse a label selector string into a map[string]string
+func ParseStringToLabels(in string) map[string]string {
+	labels := map[string]string{}
+
+	if in == "" {
```

Comment:
This check shouldn't be necessary. The `for` loop will just be a no-op if the string is empty.

Reply:
Interesting, `strings.Split()` always returns at least one element even for an empty string, see https://play.golang.org/p/JmwFMWs7u9a

Reply:
Ah, I was thinking it would be empty, my memory failed me. That really is unexpected to me.